### PR TITLE
fix: parse some options to boolean

### PR DESCRIPTION
**Pull request by aaqilniz**
This PR is related to [this](https://github.com/loopbackio/loopback-next/pull/9128) PR at loopback-next. The values received from lb4 cli for the options: treatCHAR1AsString, treatBIT1AsBit, and treatTINYINT1AsTinyInt are strings. Here they are being changed to boolean values.

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [x] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

### CODE REVIEW

**Comment by aaqilniz** [comment]
Is it possible to test the connector with a scenario where the values passed to it are actually from the `lb4 discover`? I am asking this because there is already a test present [here](https://github.com/loopbackio/loopback-connector-mysql/blob/master/test/mysql.discover.test.js#L346) which tests these options but they don't cover the values passed from lb4 discover (which will be strings).

**Comment by aaqilniz** [comment]
Hi @achrinza, can you have a look at this PR too? Thanks.

**Review by achrinza — APPROVED** [review]

**Review by samarpanB — APPROVED** [review]

**Comment by aaqilniz** [comment]
Hi, @dhmlau, can you please have a look at this PR?

**Comment by aaqilniz** [comment]
Hi, @dhmlau. I have added the tests. Please have a look.

**Comment by samarpanB** [comment]
Checks will be fixed after merging #482

**Comment by aaqilniz** [comment]
Hi, @dhmlau. The tests were failing due to a few linting issues in the tests I wrote. I have just fixed them. Can you please have a look at the PR?

**Comment by aaqilniz** [comment]
Hi, @dhmlau. Can you please have a look at the PR? Two tests are failing due to access being denied to the database.

**Comment by samarpanB** [comment]
@aaqilniz can you rebase this PR ?

**Comment by aaqilniz** [comment]
> @aaqilniz can you rebase this PR ?

Tried rebasing but everything is up to date.

**Comment by aaqilniz** [comment]
Hi, @dhmlau. Can you please have a look at the PR?

**Comment by dhmlau** [comment]
@samarpanB, do you know why CI is failing?

**Comment by dhmlau** [comment]
i've created a test PR https://github.com/loopbackio/loopback-connector-mysql/pull/483.

**Comment by aaqilniz** [comment]
Hi, @dhmlau. I think tests for both PRs (#479 & #483) are failing for different reasons.

**Comment by aaqilniz** [comment]
Hi, @dhmlau . Can you have a look at this PR, too? 

Thanks.

**Review by dhmlau — APPROVED** [review]

**Comment by aaqilniz** [comment]
Hi, @dhmlau. Due to failed tests, the merge button is disabled for me. I am not sure how should I proceed to merge the PR.

**Comment by samarpanB** [comment]
@aaqilniz We need to first work on this - https://github.com/loopbackio/loopback-connector-mysql/issues/477.

**Comment by samarpanB** [comment]
Started a draft PR. Tests need to be updated actually. Working on that.

**Comment by dhmlau** [comment]
_I posted a message earlier but it didn't seem to show up. Reposting_

Let's merge it and address the CI issue separately. Thanks.

**Pull request merged**
